OpenEdXMobile/src/main/java/org/edx/mobile/module/prefs/PrefBaseManager.kt

```diff
@@ -0,0 +1,205 @@
+        sharedPreferences.getString(key, defaultValue) ?: defaultValue
+
+    /**
+     * Returns boolean value for the given key, defaultValue if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return boolean
+     */
+    fun getBoolean(key: String, defaultValue: Boolean): Boolean =
+        sharedPreferences.getBoolean(key, defaultValue)
+
+    /**
+     * Returns long value for the given key, defaultValue if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return long
+     */
+    fun getLong(key: String, defaultValue: Long = -1): Long =
+        sharedPreferences.getLong(key, defaultValue)
+
+    /**
+     * Returns float value for the given key, defaultValue if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return float
+     */
+    fun getFloat(key: String, defaultValue: Float = -1f): Float =
+        sharedPreferences.getFloat(key, defaultValue)
+
+    /**
+     * Returns int value for the given key, defaultValue if no value is found.
+     *
+     * @param key
+     * @param defaultValue
+     * @return int
+     */
+    fun getInt(key: String, defaultValue: Int = -1): Int =
+        sharedPreferences.getInt(key, defaultValue)
+
+    /**
+     * Remove the key and its associated value.
+     *
+     * @param key
+     */
+    fun removeKey(key: String) = editor.remove(key).apply()
+
+    /**
+     * Remove all key value pairs
+     */
+    fun clear() = editor.clear().apply()
+
+    /**
+     * Migrates data from the old preference manager to the new preference manager and remove the old preferences
+     *
+     * @param oldPrefBaseManager The old preference manager to migrate data from.
+     */
+    fun migrateData(oldPrefBaseManager: PrefBaseManager) {
+        oldPrefBaseManager.sharedPreferences.all.forEach { (key, value) ->
+            when (value) {
+                is Int -> put(key, value)
+                is Long -> put(key, value)
+                is Float -> put(key, value)
+                is Boolean -> put(key, value)
+                is String -> put(key, value)
+            }
+            oldPrefBaseManager.removeKey(key)
+        }
+    }
+
+    companion object {
+
+        const val APP_FEATURES = "app_features"
+        const val APP_INFO = "pref_app_info"
+        const val LOGIN = "pref_login"
+        const val USER = "pref_user"
+        const val WIFI = "pref_wifi"
+        const val VIDEOS = "pref_videos"
+        const val COURSE_CALENDAR_PREF = "course_calendar_pref"
+
+        const val DEFAULT_VALUE = "NONE"
+
+        private val preferencesList = listOf(
+            APP_FEATURES,
+            APP_INFO,
+            LOGIN,
+            USER,
+            WIFI,
+            VIDEOS,
+            COURSE_CALENDAR_PREF
+        )
+
+        val allPreferenceFileNames: Array<String>
+            get() = preferencesList.map { "$it.xml" }.toTypedArray()
```

Comment:
We can use it directly like this because all values are constant: 

```
val allPreferenceFileNames = preferencesList.map { "$it.xml" }.toTypedArray()